src/core/Tree/TreeNode.tsx

```diff
@@ -88,24 +68,34 @@ export const TreeNode = (props: TreeNodeProps) => {
     [depthLevel, style],
   );
 
-  const [expanded, setExpanded] = React.useState(isExpanded);
-  const [active, setActive] = React.useState(isActive);
+  const context = React.useContext(TreeContext);
+  const [expanded, setExpanded] = React.useState(false);
+  const isSelected = React.useMemo(() => {
+    return context?.selectedNode === label;
+  }, [context?.selectedNode, label]);
 
   return (
     <li role='treeitem' aria-expanded={expanded} {...rest}>
       <div
         className={cx('iui-tree-node', {
-          'iui-active': active,
+          'iui-active': isSelected,
           'iui-disabled': isDisabled,
           className,
         })}
         style={style_level}
-        onClick={() => {
-          setActive(!active);
-          onSelected?.();
+        onClick={(e) => {
+          if (e.currentTarget != e.target) {
```

Comment:
Why you need this check?

Reply:
This check prevents the node from being selected when clicking on the expander button (without this it was expanding and selecting when clicking the button). However, I noticed that now this doesn't select the node when clicking over the label/sublabel so I will need to rethink this

Reply:
Oh, in that case you need to add `e.stopPropagation()` in expander `onClick` handler.